src/main/java/com/jfrog/ide/idea/ui/filters/FilterManagerService.java

```diff
@@ -1,16 +1,79 @@
 package com.jfrog.ide.idea.ui.filters;
 
+import com.intellij.openapi.application.ApplicationManager;
+import com.intellij.openapi.components.PersistentStateComponent;
 import com.intellij.openapi.components.ServiceManager;
+import com.intellij.openapi.components.State;
 import com.intellij.openapi.project.Project;
+import com.intellij.util.messages.MessageBus;
 import com.jfrog.ide.common.filter.FilterManager;
+import com.jfrog.ide.idea.events.ApplicationEvents;
 import org.jetbrains.annotations.NotNull;
+import org.jfrog.build.extractor.scan.License;
+import org.jfrog.build.extractor.scan.Severity;
+
+import java.util.HashMap;
+import java.util.Map;
 
 /**
  * @author yahavi
  */
-public class FilterManagerService extends FilterManager {
+@State(name = "FilterState")
+public class FilterManagerService extends FilterManager implements PersistentStateComponent<FilterManagerService.FiltersState> {
+
+    private FiltersState state;
 
     public static FilterManager getInstance(@NotNull Project project) {
         return ServiceManager.getService(project, FilterManagerService.class);
     }
+
+    /**
+     * Only on first scan after project opens, update the selected-licenses from state.
+     * After updating, set state's licenses map to null.
+     * @return Selected licenses map according to persisted state.
+     */
+    @Override
+    public Map<License, Boolean> getSelectedLicenses() {
+        Map<License, Boolean> selectedLicenses = super.getSelectedLicenses();
+        if (state.selectedLicences != null) {
+            for (License license : selectedLicenses.keySet()) {
+                if (state.selectedLicences.containsKey(license.getName())) {
+                    selectedLicenses.put(license, state.selectedLicences.get(license.getName()));
+                }
+            }
+            state.selectedLicences = null;
+            // Update licenses tree with applied filters.
+            MessageBus messageBus = ApplicationManager.getApplication().getMessageBus();
+            messageBus.syncPublisher(ApplicationEvents.ON_SCAN_FILTER_LICENSES_CHANGE).update();
+        }
+        return selectedLicenses;
+    }
+
+    static class FiltersState {
+        public Map<Severity, Boolean> selectedSeverities;
+        public Map<String, Boolean> selectedLicences;
+    }
+
+    @Override
+    public FiltersState getState() {
+        FiltersState state = new FiltersState();
+        state.selectedSeverities = getSelectedSeverities();
+
+        // Persist only license name and whether it is selected or not.
+        Map<License, Boolean> selectedLicences = super.getSelectedLicenses();
+        state.selectedLicences = new HashMap<>();
+        for (License license : selectedLicences.keySet()) {
+            state.selectedLicences.put(license.getName(), selectedLicences.get(license));
+        }
+
+        return state;
+    }
+
+    public void loadState(FiltersState state) {
+        if (state == null) {
+            return;
+        }
```

Comment:
```suggestion
```
I think the state object can't be null, because Intellij should call it only if there is a state to load. This is described also [here](https://github.com/JetBrains/intellij-community/blob/webstorm/193.5233.13/platform/projectModel-api/src/com/intellij/openapi/components/PersistentStateComponent.java#L40).